stimproc/testing/netstimtester.py

```diff
@@ -0,0 +1,260 @@
+            start = time()
+            data = self.conn.recv(1024)
+            end = time()
+            responseTime = (end - start) * 1000  # calculates response time in ms rounded to 3 decimals
+
+            logging.info("Recv: " + data.decode().strip())
+            logging.info("Response Time: " + str(round(responseTime, 3)) + " ms")
+            logging.info("Threshold: " + str(command[1]) + " ms")
+
+            # Determines if response time measured is within threshold
+            if responseTime <= command[1]:
+                withinThreshold = True
+
+            else:
+                withinThreshold = False
+
+            logging.info("Within Threshold: " + str(withinThreshold))
+
+            # Determines if response matches the expected response
+            if data.strip().split(b',')[0] == command[2]:
+                expectedResponse = True
+            else:
+                expectedResponse = False
+
+            logging.info("Expected Response: " + str(expectedResponse))
+
+            # Determines if the test is a pass or failure - note that expected errors constitute passes
+            if withinThreshold and expectedResponse:
+                self.passes += 1
+                self.responses.append(responseTime)
+                self.evalResponses(command[0], responseTime)
+                self.testCSV.append(["Pass", responseTime, command[0].decode(), data.strip().decode()])
+                self.printResult("Pass", True, False)
+
+            else:
+                self.fails += 1
+                testPass = False  # Test fails if a command fails
+                self.testCSV.append(["Fail", responseTime, command[0].decode(), data.strip().decode()])
+                self.printResult("Fail", False, False)
+
+            self.lineSplit()
+
+            sleep(0.25)  # delay between each command sent - this can be changed
+
+        # Determines if the test passes overall - one command failure will result in a test failure
+        if testPass:
+            self.testPasses += 1
+            self.printResult("Test Passed", True, True)
+
+        else:
+            self.testFails += 1
+            self.printResult("Test Failed", False, True)
+
+        self.lineSplit()
+
+    def run(self):
+
+        self.runTests()
+
+        logging.info("Command Passes: " + str(self.passes) + "\t\t\tCommand Failures: " + str(self.fails))
+        logging.info("Test Passes: " + str(self.testPasses) + "\t\t\t\tTest Failures: " + str(self.testFails))
+
+        self.lineSplit()
+
+        logging.info("All Responses:")
+        self.findStats(self.responses)
+        logging.info("\nStim Responses:")
+        self.findStats(self.stimResponses)
+        logging.info("\nConfig Responses:")
+        self.findStats(self.configResponses)
+        logging.info("\nThetaConfig Responses:")
+        self.findStats(self.thetaConfigResponses)
+
+        self.writeCSV()  # writes the csv file
+
+
+if __name__ == '__main__':
+    tester = Tester()
```

Comment:
We should have here the ability to optionally override the default host and port to listen on using the command line.  This will be helpful for several types of testing which should be doable without modifying this testing code.